app/databrowser/src/main/java/org/csstudio/trends/databrowser3/ui/Perspective.java

```diff
@@ -196,11 +202,23 @@ private void createContextMenu()
             items.add(new SeparatorMenuItem());
             items.add(new PrintAction(plot.getPlot()));
             items.add(new SaveSnapshotAction(plot.getPlot()));
-            if (EmailPreferences.isEmailSupported())
-                items.add(new SendEmailAction(this, Messages.ActionEmailTitle, Messages.ActionEmailBody, () ->  plot.getPlot().getImage()));
+
+            SelectionService.getInstance().setSelection(this, Arrays.asList(DatabrowserSelection.of(model, plot)));
+            List<ContextMenuEntry> supported = ContextMenuService.getInstance().listSupportedContextMenuEntries();
+            supported.stream().forEach(action -> {
+                MenuItem menuItem = new MenuItem(action.getName(), new ImageView(action.getIcon()));
+                menuItem.setOnAction((e) -> {
+                    try {
+                        action.call(plot.getPlot(), SelectionService.getInstance().getSelection());
+                    } catch (Exception e1) {
+                        e1.printStackTrace();
```

Comment:
Instead of `e1.printStackTrace()`, I would suggest `logger.log(Level.WARNING, "XYZ Failed", e1)`.
The printout can easily get lost.
Using the logger means we can control the level and destination of the message, there's hopefully a somewhat useful specific message to describe the problem, plus you still get the stack trace.
Tried to document that in https://github.com/ControlSystemStudio/phoebus/pull/1031